src/main/java/frc/robot/systems/ElevatorArmFSM.java

```diff
@@ -241,52 +249,139 @@ private FSMState nextState(TeleopInput input) {
 	/* ------------------------ FSM state handlers ------------------------ */
 	/**
 	 * Handle behavior in START_STATE.
+	 *
 	 * @param input Global TeleopInput if robot in teleop mode or null if
-	 *        the robot is in autonomous mode.
+	 *              the robot is in autonomous mode.
 	 */
 	private void handleIdleState(TeleopInput input) {
-		pidControllerArm.setReference(currentEncoder, CANSparkMax.ControlType.kPosition);
+		// pidControllerArm.setReference(currentEncoder,
+		// CANSparkMax.ControlType.kPosition);
+		pidControllerArm.setReference(0, CANSparkMax.ControlType.kDutyCycle);
 	}
+
 	private void handleHighState(TeleopInput input) {
-		pidControllerArm.setReference(HIGH_ENCODER_ROTATIONS, CANSparkMax.ControlType.kPosition);
+		//pidControllerArm.setReference(HIGH_ENCODER_ROTATIONS, CANSparkMax.ControlType.kPosition);
+		armMotor.set(pid(armMotor.getEncoder().getPosition(), HIGH_ENCODER_ROTATIONS));
+		System.out.println("In high state");
 	}
+
 	private void handleMiddleState(TeleopInput input) {
-		pidControllerArm.setReference(MID_ENCODER_ROTATIONS, CANSparkMax.ControlType.kPosition);
+		//pidControllerArm.setReference(MID_ENCODER_ROTATIONS, CANSparkMax.ControlType.kPosition);
+		armMotor.set(pid(armMotor.getEncoder().getPosition(), MID_ENCODER_ROTATIONS));
+		System.out.println("In mid state");
 	}
+
 	private void handleLowState(TeleopInput input) {
-		pidControllerArm.setReference(LOW_ENCODER_ROTATIONS, CANSparkMax.ControlType.kPosition);
+		//pidControllerArm.setReference(LOW_ENCODER_ROTATIONS, CANSparkMax.ControlType.kPosition);
+		armMotor.set(pid(armMotor.getEncoder().getPosition(), LOW_ENCODER_ROTATIONS));
+		System.out.println("In low state");
 	}
+
 	private void handleMovingState(TeleopInput input) {
-		pidControllerArm.setReference(-input.getLeftJoystickY() / three,
-			CANSparkMax.ControlType.kDutyCycle);
+		System.out.println("Zeroed: " + zeroed);
+		if (!zeroed) {
+			System.out.println("here");
+			pidControllerArm.setReference(-input.getLeftJoystickY() / JOYSTICK_CONSTANT,
+				CANSparkMax.ControlType.kDutyCycle);
+			return;
+		}
+		if ((armMotor.getEncoder().getPosition() >= LOW_ENCODER_ROTATIONS
+				&& armMotor.getEncoder().getPosition() <= HIGH_ENCODER_ROTATIONS)) {
+
+			pidControllerArm.setReference(-input.getLeftJoystickY() / JOYSTICK_CONSTANT,
+						CANSparkMax.ControlType.kDutyCycle);
+
+		} else if (armMotor.getEncoder().getPosition() < LOW_ENCODER_ROTATIONS) {
+
+			if (-input.getLeftJoystickY() > 0) {
+				System.out.println("OVER ");
+				pidControllerArm.setReference(-input.getLeftJoystickY() / JOYSTICK_CONSTANT,
+						CANSparkMax.ControlType.kDutyCycle);
+			} else {
+				pidControllerArm.setReference(0, CANSparkMax.ControlType.kDutyCycle);
+			}
+
+		} else {
+			if (-input.getLeftJoystickY() < 0) {
```

Comment:
It's impossible to reach this else branch right?

Reply:
When one is true and the other is false, that's when the limit switch either goes from triggered to not triggered or vice versa. Thats the else condition and also when we zero the elevator encoder.